server/genquery2/src/genquery2_sql.cpp

```diff
@@ -0,0 +1,1172 @@
+            "with{recursive_op} cte_drh as ("
+                "select "
+                    "resc_id, "
+                    "resc_name hier, "
+                    "case "
+                        "when resc_parent = '' then 0 "
+                        "else cast(resc_parent as {int_type}) "
+                    "end parent_id "
+                "from R_RESC_MAIN "
+                "where resc_id > 0 "
+                "union all "
+                "select "
+                    "cte_drh.resc_id, "
+                    "cast(concat(concat(U.resc_name, ';'), cte_drh.hier) as {char_type}(250)), "
+                    "case "
+                        "when U.resc_parent = '' then 0 "
+                        "else cast(U.resc_parent as {int_type}) "
+                    "end parent_id "
+                "from cte_drh "
+                "inner join R_RESC_MAIN U on U.resc_id = cte_drh.parent_id"
+            ") ";
+        // clang-format on
+
+        // See https://modern-sql.com/caniuse/cast_as_bigint to understand why the data types
+        // for MySQL and Oracle were chosen.
+
+        if (_database == "mysql") {
+            return fmt::format(data_resc_hier_with_clause,
+                               fmt::arg("recursive_op", " recursive"),
+                               fmt::arg("int_type", "signed"),
+                               fmt::arg("char_type", "char"));
+        }
+
+        if (_database == "oracle") {
+            return fmt::format(data_resc_hier_with_clause,
+                               fmt::arg("recursive_op", ""),
+                               fmt::arg("int_type", "integer"),
+                               fmt::arg("char_type", "varchar"));
+        }
+
+        // Assume the database supports the BIGINT data type.
+        // This is the preferred data type, is fully supported by PostgreSQL, and
+        // is defined in ISO/IEC 9075:2016-2.
+        return fmt::format(data_resc_hier_with_clause,
+                           fmt::arg("recursive_op", " recursive"),
+                           fmt::arg("int_type", "bigint"),
+                           fmt::arg("char_type", "varchar"));
+    } // generate_with_clause_for_data_resc_hier
+
+    auto generate_limit_clause(const irods::experimental::api::genquery2::options& _opts,
+                               const std::string_view _number_of_rows) -> std::string
+    {
+        if (!_number_of_rows.empty()) {
+            if (_opts.database == "mysql") {
+                return fmt::format(" limit {}", _number_of_rows);
+            }
+
+            return fmt::format(" fetch first {} rows only", _number_of_rows);
+        }
+
+        if (_opts.database == "mysql") {
+            return fmt::format(" limit {}", _opts.default_number_of_rows);
+        }
+
+        return fmt::format(" fetch first {} rows only", _opts.default_number_of_rows);
+    } // generate_limit_clause
+} // anonymous namespace
+
+namespace irods::experimental::api::genquery2
```

Comment:
Same here... Should this be `irods::experimental::genquery2`?

Reply:
Done